File: Java/HLC02_EjemplosCodigo10_14/10_Ud2IntentExplicito01/app/src/main/java/com/pms/a10_ud2intentexplicito01/SegundaActivity.java
package com.pms.a10_ud2intentexplicito01;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class SegundaActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_segunda);

        //referencia a la View TextView del layout XML
        TextView txtMensaje = (TextView)findViewById(R.id.TxtMensaje);

        //obtiene la información pasada por la activity llamadora (MainActivity) en un objeto Bundle
        Bundle bundle = getIntent().getExtras();

        //visualiza la información pasada extrayendo el valor asociado a la clave
        txtMensaje.setText("Hola " + bundle.getString("NOMBRE"));
    }
}

File: Kotlin/HLC02_EjemplosCodigo10_14/10_Ud2IntentExplicito01/app/src/main/java/com/pms/a10_ud2intentexplicito01/SegundaActivity.kt
package com.pms.a10_ud2intentexplicito01

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.pms.a10_ud2intentexplicito01.R
import android.widget.TextView

class SegundaActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_segunda)

        //referencia a la View TextView del layout XML
        val txtMensaje = findViewById<View>(R.id.TxtMensaje) as TextView

        //obtiene la información pasada por la activity llamadora (MainActivity) en un objeto Bundle
        val bundle = intent.extras

        //visualiza la información pasada extrayendo el valor asociado a la clave
        txtMensaje.text = "Hola " + bundle.getString("NOMBRE")
    }
}